runtime/domain.c

```diff
@@ -1267,6 +1268,13 @@ CAMLprim value caml_domain_spawn(value callback, value term_sync)
   if (caml_debugger_in_use)
     caml_fatal_error("ocamldebug does not support spawning multiple domains");
 #endif
+
+  /* When domain 0 first spawns a domain, the backup thread is not
+     active, we ensure it is started here. We do this before the first
+     domain is spawned in the event it should fail. */
+  value res = install_backup_thread_exn(domain_self);
+  if (Is_exception_result(res)) caml_raise(Extract_exception(res));
```

Comment:
On first read of this comment I thought that you were talking about the backup thread for the domain that is about to be created -- and it seemed odd to me to create it from the parent.  I wonder if the comment should be rephrased slightly to make it clear that we are talking about the backup thread of domain 0?